src/benchmark.cc

```diff
@@ -592,8 +591,13 @@ size_t RunSpecifiedBenchmarks(BenchmarkReporter* display_reporter,
   }
 
   if (FLAGS_benchmark_list_tests) {
-    for (auto const& benchmark : benchmarks)
-      Out << benchmark.name().str() << "\n";
+    if (FLAGS_benchmark_format == "json") {
```

Comment:
you don't need to do this.  at this point we've already determined the right reporter type and it's stored in `display_reporter`.  just calling `display_reporter->List(benchmarks)` should be sufficient.

otherwise why is it virtual? :)